lib/faraday/request/json.rb

```diff
@@ -24,7 +24,13 @@ def on_request(env)
       private
 
       def encode(data)
-        ::JSON.generate(data)
+        if options[:encoder].is_a?(Array) && options[:encoder].size >= 2
+          options[:encoder][0].public_send(options[:encoder][1], data)
+        elsif options[:encoder].respond_to?(:encode)
+          options[:encoder].encode(data)
+        else
+          ::JSON.generate data
+        end
```

Comment:
This should be good from a performance point of view considering we don't do any instantiation